catalyst-gateway/bin/src/service/api/documents/post_document_index_query/response.rs

```diff
@@ -241,29 +245,66 @@ impl Example for IndexedDocumentVersionDocumented {
     }
 }
 
-impl TryFrom<SignedDocBody> for IndexedDocumentVersionDocumented {
+pub(crate) const NEWER_DOC_ON_DEPRECATED_ENDPOINT: &'static str =
+    "NEWER_DOC_ON_DEPRECATED_ENDPOINT";
+
+impl TryFrom<FullSignedDoc> for IndexedDocumentVersionDocumented {
     type Error = anyhow::Error;
 
-    fn try_from(doc: SignedDocBody) -> Result<Self, Self::Error> {
+    fn try_from(doc: FullSignedDoc) -> Result<Self, Self::Error> {
+        // should allow deprecated only for this response
+        if !CatalystSignedDocument::try_from(doc.raw())?.is_deprecated()? {
```

Comment:
after filtering you could assume that at this place you will get only deprecated documents only